.github/workflows/ci.yaml

```diff
@@ -0,0 +1,159 @@
+          path: ./demo
+
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version-file: .nvmrc
+          cache: npm
+
+      - name: Install dependencies
+        run: npm ci --ignore-scripts
+
+      - name: Lint
+        run: npm run lint:ci
+
+  test:
+    needs: [build]
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version-file: .nvmrc
+          cache: npm
+
+      - name: Install dependencies
+        run: npm ci --ignore-scripts
+
+      - name: Download dist artifacts
+        uses: actions/download-artifact@v4
+        with:
+          name: dist-artifact
+          path: packages
+
+      - name: Test
+        run: npm run test:ci
+
+      - name: Coveralls
+        if: env.COVERALLS_REPO_TOKEN != ''
+        run: npm exec -- coveralls < .cache/coverage/lcov.info
+        env:
+          COVERALLS_REPO_TOKEN: ${{ secrets.COVERALLS_REPO_TOKEN }}
+
+  deploy:
+    needs: [build-demo, lint, test]
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version-file: .nvmrc
+          cache: npm
+
+      - name: Install dependencies
+        run: npm ci --ignore-scripts
+
+      - name: Download demo artifacts
+        uses: actions/download-artifact@v4
+        with:
+          name: demo-artifact
+          path: ./demo
+
+      - name: Deploy
+        run: utils/scripts/deploy.mjs
+        env:
+          NETLIFY_TOKEN: ${{ secrets.NETLIFY_TOKEN }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          NETLIFY_SITE_ID: ${{ secrets.NETLIFY_SITE_ID }}
+          # Maintain compatibility with @zendeskgarden/scripts
+          CIRCLE_PROJECT_USERNAME: zendeskgarden
+          CIRCLE_PROJECT_REPONAME: react-components
+          CIRCLECI: true
+
+  publish:
+    needs: [build, lint, test]
+    runs-on: ubuntu-latest
+    if: github.ref == 'refs/heads/main'
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0 # Lerna requires the full history, including tags
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version-file: .nvmrc
+          cache: npm
+          registry-url: https://registry.npmjs.org # Sets the registry in the project level .npmrc
+
+      - name: Install dependencies
+        run: npm ci --ignore-scripts
+
+      - name: Download dist artifacts
+        uses: actions/download-artifact@v4
+        with:
+          name: dist-artifact
+          path: packages
+
+      # - name: Publish to npm
+      #   run: npm exec -- lerna publish from-git --ignore-scripts --yes
+      #   env:
+      #     NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
```

Comment:
The million 💰 question. Will this work? 🤷‍♂️ 

Based on my [research](https://github.com/actions/checkout/issues/217#issuecomment-634802933) `fetch-depth: 0` is all we need for `lerna publish` to work.

> [!NOTE]
> Fetching the entire Git history + tags takes [~2min](https://github.com/zendeskgarden/react-components/actions/runs/9462583243/job/26066351150#step:2:52).


Do you have any recommendations on how to test this?

Reply:
For the smoke test, let's run `npm whoami`. If the result comes back with `zendesk-garden`, then we have confidence that the configuration is solid.